### PR TITLE
test: Pocketic migration tests

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
We would like to increase the variety of events that occur in migration tests. For example:
- The e2e migration tests do not add accounts during the migration.
- The e2e tests do not cover redeploying in the middle of a migration.  This has been tested manually but not in CI.

# Changes
- Separate running unit tests from running integration tests in CI.
  - Note: Historically we had only unit tests.  The new tests are integration tests.  If we do not separate them, CI becomes significantly slower.
- Add a `pocket-ic` test environment with a stable reference `nns-dapp` and a second `nns-dapp` to which we apply migrations.
  - Note: The two canisters should contain the same data regardless of what migrations, rollbacks or account changes we apply.
- Use the `pocket-ic` test environemnt to verify that accounts are not lost during a normal migration.
- Use the `pocket-ic` test environment to verify that accounts are not lost if we need to roll back.
- Use `proptest` to apply many random operations to the test environment and verify that, in doing so, no accounts are lost.

# Tests
- These new tests are exercised in CI.

# Todos

- [ ] Add entry to changelog (if necessary).

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -68,10 +68,10 @@ jobs:
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}-1
       - name: Lint rust code
         run: ./scripts/lint-rs
-      - name: Run Tests
+      - name: Run Unit Tests
         shell: bash
         run: |
-          cargo test --all-targets --all-features
+          cargo test --lib --all-targets --all-features
```
Comment (bitdivine) [inline .github/workflows/checks.yml:74]: Integration tests require the Wasms to have been built for testing.  There is no point in delaying unit tests until that is done.  Also running the Wasm build and unit tests sequentially would make CI significantly slower.  So we run the unit tests in parallel to the docker build and run the integration tests after the docker build is complete.

File: rs/backend/tests/migration.rs [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+///
+/// Note: u8 is used for some variants, rather than the larger types supported by the actual API, to keep the test runtime reasonable.
+#[derive(Debug, Eq, PartialEq, EnumIter, Copy, Clone)]
+enum Operation {
+    UpgradeToMap,
+    UpgradeToStableStructures,
+    Tick(u8),
+    CreateToyAccounts(u8),
+    // TODO: Modify a random account
+    // TODO: Delete a random account
```
Comment (bitdivine) [inline rs/backend/tests/migration.rs:157]: Note: There is a dedicated test method that makes it easy to create accounts, so we use that to modify the accounts database.  Modifying and deleting accounts is more involved and is to be done in separate PRs, as this PR is already too big.

**Comment by bitdivine** [comment]
Note: CI failed on some front-end tests unrelated to this PR.  Updating to head of  main; hopefully the error has either been fixed or is transient.

**Review by jasonz-dfinity — APPROVED** [review]
